src/content/docs/more-integrations/open-source-telemetry-integrations/opentelemetry/opentelemetry-tutorials/opentelemetry-tutorial-java.mdx

```diff
@@ -0,0 +1,1132 @@
+        public class Application {
+
+            private static volatile OpenTelemetry openTelemetry = OpenTelemetry.noop();
+
+            public static void main(String[] args) {
+                // Build the SDK auto-configuration extension module
+                OpenTelemetrySdk openTelemetrySdk = AutoConfiguredOpenTelemetrySdk.builder()
+                        .setResultAsGlobal(false)
+                        .build()
+                        .getOpenTelemetrySdk();
+                Application.openTelemetry = openTelemetrySdk;
+
+                // Set GlobalLoggerProvider, which is used by Log4j2 appender
+                GlobalLoggerProvider.set(openTelemetrySdk.getSdkLoggerProvider());
+
+                // Register runtime metrics instrumentation
+                BufferPools.registerObservers(openTelemetrySdk);
+                Classes.registerObservers(openTelemetrySdk);
+                Cpu.registerObservers(openTelemetrySdk);
+                GarbageCollector.registerObservers(openTelemetrySdk);
+                MemoryPools.registerObservers(openTelemetrySdk);
+                Threads.registerObservers(openTelemetrySdk);
+
+                SpringApplication.run(Application.class, args);
+            }
+
+            @Bean
+            public OpenTelemetry openTelemetry() {
+                return openTelemetry;
+            }
+
+            // Add Spring WebMVC instrumentation by registering a tracing filter
+            @Bean
+            public Filter webMvcTracingFilter(OpenTelemetry openTelemetry) {
+                return SpringWebMvcTelemetry.create(openTelemetry).createServletFilter();
+            }
+        }
+        ```
+
+3. Create a directory called `resources` in `Uninstrumented/src/main`.
+4. In this new directory, create a file called `log4j2.xml` with the following contents:
+        ```xml
+        <?xml version="1.0" encoding="UTF-8"?>
+        <Configuration status="WARN" packages="io.opentelemetry.instrumentation.log4j.appender.v2_17">
+          <Appenders>
+            <Console name="ConsoleAppender" target="SYSTEM_OUT" follow="true">
+              <PatternLayout pattern="%d{yyyy-mm-dd HH:mm:ss.SSS} [%t] %-5level %logger{36} - %msg%n"/>
+            </Console>
+            <OpenTelemetry name="OpenTelemetryAppender" />
+          </Appenders>
+          <Loggers>
+            <Root level="info">
+              <AppenderRef ref="OpenTelemetryAppender" />
+              <AppenderRef ref="ConsoleAppender" />
+            </Root>
+          </Loggers>
+        </Configuration>
+        ```
+    <Callout variant="tip">
+      The `packages=...` section of this line allows Log4J to find and configure the `OpenTelemetryAppender`. The source code is [here] (https://github.com/open-telemetry/opentelemetry-java-instrumentation/tree/main/instrumentation/log4j/log4j-appender-2.17/library) and was added as a dependency via `io.opentelemetry.instrumentation:opentelemetry-log4j-appender-2.17`)
```

Comment:
note about descriptive url links

Reply:
@zuluecho9 I revised it to this: The source code is [in the OpenTelemetry repository]....